crates/gen-c/src/lib.rs

```diff
@@ -1056,6 +1056,11 @@ impl Generator for C {
     fn finish(&mut self, iface: &Interface, files: &mut Files) {
         self.src.h(&format!(
             "\
+                #ifdef __cplusplus
+                extern \"C\"
+                {{
+                #endif
+
```

Comment:
Could this go inside the `#ifndef __BINDINGS_{0}_H` guard below? That makes it easier for C++ compilers to recognize the include-guard idiom and avoid opening a header multiple times.